### PR TITLE
Without -AconservativeUninferredTypeArguments, NEVER issue error about uninferred type arguments

### PR DESCRIPTION
**Pull request by kelloggm**
This is an extension to the [hack](https://github.com/typetools/checker-framework/issues/979#issuecomment-279578221) that prevents most errors related to #979, which did not prevent the CF from issuing errors involving method references (and possibly in other scenarios we don't know about). The test case in this PR was provided by @msridhar in a [comment](https://github.com/typetools/checker-framework/issues/3001#issuecomment-570026011) on #3001.

The technique I'm using is to check the message to be printed in the error reporter, and not issue it if it includes the text about inference failing added by [`DefaultAnnotatedTypeFormatter`](https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/type/DefaultAnnotatedTypeFormatter.java#L389) to wildcards for which inference has failed, which @mernst suggested.

### CODE REVIEW

**Comment by msridhar** [comment]
FYI I also opened #3032 for the same test case.  If there is a proper fix for that case that doesn't require fixing #979, we may want to use a different test for this change

**Comment by smillst** [comment]
@kelloggm all-tests is failing.  Is that expected?

**Comment by kelloggm** [comment]
@smillst It was not! Sorry about that. The problem was that my original solution didn't take into account that the tests run with `-Anomsgtext`. It should be fixed now to account for that case.

**Comment by mernst** [comment]
@kelloggm all_tests seems to still be failing.

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 1]
```diff
@@ -1358,6 +1359,29 @@ private void message(
...
+        if (this.processingEnv.getOptions() != null
+                && !this.processingEnv
+                        .getOptions()
+                        .containsKey("conservativeUninferredTypeArguments")) {
+
+            if (messageText.contains("INFERENCE FAILED")) {
+                return;
+            } else if (this.processingEnv.getOptions().containsKey("nomsgtext")) {
+                for (Object arg : args) {
+                    if (arg != null && arg instanceof String) {
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: The two null checks for `arg` are redundant, as `instanceof` is only true for non-null values.

**Comment by kelloggm** [comment]
@smillst I eventually decided to leave this open and directly work around the problem I showed you this morning. It's imperfect, but I think it does what we want it to for the most part, and I've left a pretty detailed comment of my reasoning in the code.

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java [hunk 2]
```diff
@@ -681,6 +682,36 @@ public void clearAnnotations() {
         annotations.clear();
     }
 
+    /**
+     * @return whether this type or any component type is a wildcard type for which Java 7 type
+     *     inference is insufficient. See issue 979, or the documentation on AnnotatedWildcardType.
+     */
+    public boolean containsUninferredTypeArguments() {
+        return uninferredTypeArgumentScanner.visit(this);
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java]: This throws a NPE when `this` is a declared type with out any type arguments.  #3044 has a version of the visitor that works correctly.

**Comment by kelloggm** [comment]
Closing in favor of #3044, which solves the same problem in a cleaner way.

**Pull request closed without merging**